### ISSUE TITLE
False positive on new `MD051/link-fragments` rule (v0.32)

### ISSUE THREAD

**Issue by JonnyIngeno** [igorshubovych/markdownlint-cli#305]
I get errors on some of my links with the new `MD051/link-fragments` rule.

```bash
$ yarn lint
$ eslint --ext .ts .
$ ls-lint
$ markdownlint '**/*.md'
README.md:35:1 MD051/link-fragments Link fragments should be valid [Context: "[⬆️ Back to Table of Contents](#table-of-contents-)"]
README.md:41:1 MD051/link-fragments Link fragments should be valid [Context: "[⬆️ Back to Table of Contents](#table-of-contents-)"]
README.md:1 MD051/link-fragments Link fragments should be valid [Context: "[⬆️ Back to Table of Contents](#table-of-contents-)"]
README.md:183:1 MD051/link-fragments Link fragments should be valid [Context: "[⬆️ Back to Table of Contents](#table-of-contents-)"]
README.md:8:1 MD051/link-fragments Link fragments should be valid [Context: "[⬆️ Back to Table of Contents](#table-of-contents-)"]
```

I don't know what is the heuristic for the rule, but these are all valid links that work in production.

While I understand the `-` at the end is "weird", it's how Github works, I can't change that :(

**Comment by DavidAnson** [comment]
Please link to the file that generates these errors or show an example of a corresponding Markdown header. I'd like to see how this comes up in practice.

**Comment by DavidAnson** [comment]
Thanks! I'll have a look tonight.

**Comment by DavidAnson** [comment]
This is fixed in the library for the next release (likely a patch in the near future): https://github.com/DavidAnson/markdownlint/commit/1bd748f79e243c4e8afe2446280b3cdce671934f

**Comment by JonnyIngeno** [comment]
Thank you very much!!!

**Comment by DavidAnson** [comment]
https://github.com/igorshubovych/markdownlint-cli/releases/tag/v0.32.1

**Comment by JonnyIngeno** [comment]
@DavidAnson 

Sorry to bother you, but I can't seem to be able to retrieve the `0.32.1` release.

When checking on <npmjs.com>, it's still on `0.32.0`. Does it take more time to sync correctly or there's been a problem with the release process?

**Comment by DavidAnson** [comment]
Oop, I may have forgotten to run "npm publish". Will do so this evening, sorry about that! (This project has a slightly different process than all my others.)

**Comment by DavidAnson** [comment]
Okay, try again now, please!

**Comment by JonnyIngeno** [comment]
Yep, it's there. Thanks!
